docs/guide/troubleshooting.md

```diff
@@ -0,0 +1,94 @@
+
+To improve performance on Windows, consider using alternative file system environments:
+
+1. [**Dev Drive**](https://learn.microsoft.com/en-us/windows/dev-drive/): A newer Windows feature designed for developer workloads, using the Resilient File System (ReFS). Using a Dev Drive can lead to a **2x to 3x speedup** compared to the standard Windows NTFS file system for file system operations.
+2. [**Windows Subsystem for Linux (WSL)**](https://learn.microsoft.com/en-us/windows/wsl/): WSL lets Linux environment to run on Windows easily, which offers significantly better file system performance. Placing your project files and running the build process within WSL can result in speedups of around **10x** compared to the standard Windows NTFS file system for file system operations.
+
+:::details Benchmark Reference
+
+The benchmark script used is described in this blog post ([How fast can you open 1000 files?](https://lemire.me/blog/2025/03/01/how-fast-can-you-open-1000-files/)).
+
+The results were:
+
+|    File System / Threads |     1 |     2 |     4 |     8 |    16 |
+| -----------------------: | ----: | ----: | ----: | ----: | ----: |
+|             Windows NTFS | 286ms | 153ms |  85ms | 106ms | 110ms |
+| Windows Dev Drive (ReFS) | 124ms |  67ms |  35ms |  48ms |  55ms |
+|               WSL (ext4) |  24ms |  13ms | 7.8ms | 9.0ms |  13ms |
+
+The benchmark was ran on the following environment:
+
+- OS: Windows 11 Pro 23H2 22631.5189
+- CPU: AMD Ryzen 9 5900X
+- Memory: DDR4-3600 32GB
+- SSD: Western Digital Black SN850X 1GB
+
+:::
+
+<!-- Maybe write about macOS as well? -->
+
+### Plugins
+
+Plugins extend Rolldown's functionality, but can also introduce performance overhead.
+
+#### Plugin Hook Filters
+
+Rolldown provides a feature called **Plugin Hook Filters**. This allows you to specify precisely which modules a plugin hook should process, reducing the communication overhead between JavaScript and Rust. For detailed information on how filters work internally, refer to the [Plugin Development Guide - Hook Filters](/guide/plugin-development#plugin-hook-filters).
+
+If you are a plugin user and the plugin you use does not have hook filters specified, you can apply them by using the `withFilter` utility function exported by Rolldown.
+
+```js
+import yaml from '@rollup/plugin-yaml';
+import { defineConfig, withFilter } from 'rolldown';
+
+export default defineConfig({
+  plugins: [
+    // Run the transform hook of the `yaml` plugin only for modules which end in `.yaml`
+    withFilter(
+      yaml({
+        /*...*/
+      }),
+      { transform: { id: /\.yaml$/ } },
+    ),
+  ],
+});
+```
+
+#### Leverage Built-in Features
```

Comment:
I'm planning a dedicated section on rollup plugin "compat", as in:

* Not needed (builtin)
* Drop-in replacement (e.g. node polyfills/replace)

etc. etc.